Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -1594,6 +1595,8 @@ void WebProcess::prepareToSuspend(bool isSuspensionImminent, MonotonicTime estim
     SetForScope suspensionScope(m_isSuspending, true);
     m_processIsSuspended = true;
 
+    parentProcessConnection()->send(Messages::WebProcessProxy::UpdateMemoryUsage(memoryFootprint()), 0);
```

Comment:
This is very fragile. The UIProcess has a way to ask the memory footprint of a WebProcess but the value is only accurate if the WebProcess got a PrepareToSuspend IPC. This doesn't feel like the right way to implement this.

Also, there is a conflict in logic:
1. We purposefully don't trim memory in WebProcesses on suspension to get more warm caches when the process gets re-used from the cache, thus making the WebProcess gets larger.
2. But, we refuse to cache a process if it is too large.

This doesn't make much sense. Seems having a cached process without warm cache would still be more valuable than no cached process at all.